head	1.3;
access;
symbols
	OPENBSD_5_8:1.2.0.4
	OPENBSD_5_8_BASE:1.2;
locks; strict;
comment	@# @;


1.3
date	2015.10.03.14.48.09;	author jasper;	state dead;
branches;
next	1.2;
commitid	aYZxPqQ22E869w3o;

1.2
date	2015.05.14.18.20.34;	author jasper;	state Exp;
branches;
next	1.1;
commitid	MZU0mWD4K9KFxqcG;

1.1
date	2015.05.12.12.39.10;	author jasper;	state Exp;
branches;
next	;
commitid	OhXhzvY1LShDMezr;


desc
@@


1.3
log
@update to libpeas-1.16.0
@
text
@$OpenBSD: patch-tests_libpeas_extension-py_c,v 1.2 2015/05/14 18:20:34 jasper Exp $

From f5c7668f600f16f144c3772cc34c9fd929f98d28 Mon Sep 17 00:00:00 2001
From: Garrett Regier <garrettregier@@gmail.com>
Date: Tue, 12 May 2015 04:48:42 -0700
Subject: Fix Python tests on OpenBSD

--- tests/libpeas/extension-py.c.orig	Sat Feb 14 21:35:54 2015
+++ tests/libpeas/extension-py.c	Tue May 12 14:33:39 2015
@@@@ -180,11 +180,19 @@@@ test_extension_py_already_initialized_subprocess (void
 static void
 test_extension_py_mixed_python (void)
 {
+  GTestSubprocessFlags flags = 0;
+
+  /* Loading both Python 2 and Python 3 might cause
+   * the linker to spew warnings, i.e. on OpenBSD, so
+   * only inherit standard error when debugging
+   */
+  if (g_getenv ("PEAS_DEBUG") != NULL)
+    flags |= G_TEST_SUBPROCESS_INHERIT_STDERR;
+
   g_test_trap_subprocess (EXTENSION_TEST_NAME (PY_LOADER,
                                                "mixed-python/subprocess"),
-                          0, G_TEST_SUBPROCESS_INHERIT_STDERR);
+                          0, flags);
   g_test_trap_assert_passed ();
-  g_test_trap_assert_stderr ("");
 }
 
 static void
@


1.2
log
@committed upstream
@
text
@d1 1
a1 1
$OpenBSD: patch-tests_libpeas_extension-py_c,v 1.1 2015/05/12 12:39:10 jasper Exp $
@


1.1
log
@fix regress tests on OpenBSD; with upstream
@
text
@d1 1
a1 1
$OpenBSD$
d3 4
a6 1
https://bugzilla.gnome.org/show_bug.cgi?id=727380
@

